panoptes_client/caesar.py

```diff
@@ -68,20 +78,34 @@ def get_extracts_by_workflow_and_subject(self, workflow_id, subject_id):
         return self.http_get(
             f'workflows/{workflow_id}/extractors/extractor/extracts', params={'subject_id': subject_id})[0]
 
-    def add_workflow(self, workflow_id, public_extracts=False, public_reductions=False):
+    def save_workflow(self, workflow_id, public_extracts=False, public_reductions=False):
         """
-        Adds workflow with provided workflow_id to Caesar. Returns workflow as a dict from Caesar if successful.
+        Adds/updates workflow with provided workflow_id to Caesar. Returns workflow as a dict from Caesar if created.
 
         Examples::
             Caesar().add_workflow(123, public_extracts=True, public_reductions=True)
         """
-        return self.http_post('workflows', json={
-            'workflow': {
-                'id': workflow_id,
-                'public_extracts': public_extracts,
-                'public_reductions': public_reductions
-            }
-        })[0]
+        try:
+            self.get_workflow(workflow_id)
+        except Exception as err:
```

Comment:
it's generally bad practice to swallow generic exception types here as it might be too broad and swallow errors we aren't expecting. I.e. avoid swallow all and check error strings, rather only catch the specific ones and fail on non-expected ones.

Is there a specific error class we can catch here for the `.get_workflow` call? perhaps https://github.com/zooniverse/panoptes-python-client/blob/535d38b2ff37e01223b2e5045bd30a6a77b9d8d8/panoptes_client/panoptes.py#L1110